lib/model-interfaces/idefics/functions/request-handler/adapters/claude.py

```diff
@@ -0,0 +1,104 @@
+from .base import MultiModalModelBase
+from genai_core.types import ChatbotAction, ChatbotMessageType
+from urllib.parse import urljoin
+import os
+from genai_core.clients import get_bedrock_client
+import json
+import requests
+from base64 import b64encode
+from genai_core.registry import registry
+
+
+def get_image_message(
+    file: dict,
+):
+    img = requests.get(
+        f"{urljoin(os.environ['CHATBOT_FILES_PRIVATE_API'], file['key'])}"
+    ).content
+    return {
+        "type": "image",
+        "source": {
+            "type": "base64",
+            "media_type": "image/jpeg",
+            "data": str(b64encode(img), "ascii"),
+        },
+    }
+
+
+class Claude3(MultiModalModelBase):
+    model_id: str
+    client: any
+
+    def __init__(self, model_id: str):
+        self.model_id = model_id
+        self.client = get_bedrock_client()
+
+    def format_prompt(self, prompt: str, messages: list, files: list) -> str:
+        prompts = []
+
+        # Chat history
+        for message in messages:
+            if message.type.lower() == ChatbotMessageType.Human.value.lower():
+                user_msg = {
+                    "role": "user",
+                    "content": [{"type": "text", "text": message.content}],
+                }
+                prompts.append(user_msg)
+                message_files = message.additional_kwargs.get("files", [])
+                for message_file in message_files:
+                    user_msg["content"].append(get_image_message(message_file))
+            if message.type.lower() == ChatbotMessageType.AI.value.lower():
+                prompts.append({"role": "assistant", "content": message.content})
+
+        # User prompt
+        user_msg = {
+            "role": "user",
+            "content": [{"type": "text", "text": prompt}],
+        }
+        prompts.append(user_msg)
+        for file in files:
+            user_msg["content"].append(get_image_message(file))
+
+        return json.dumps(
+            {
+                "anthropic_version": "bedrock-2023-05-31",
+                "max_tokens": 512,
```

Comment:
are hardcoded `max_tokens` and `temperature` really needed here?

Reply:
Just replicated this: https://github.com/aws-samples/aws-genai-llm-chatbot/blob/0545d45e4a1f7f321a3247f5988da96b2b37a805/lib/model-interfaces/langchain/functions/request-handler/adapters/bedrock/base.py#L36-L37

Reply:
These values will be overridden in `handle_run`.